### PR TITLE
updates & cleanup

### PR DESCRIPTION
**Pull request by stefanpenner**
* upgrade to latest ember-cli
* remove unused files
* upgrade to some fancy ES2015 syntax
* enable ember-try

- [ ] still some test failures, i'll revisit later today.
- [ ] fix deprecations
  - [ ] remove ObjectController usage.

### CODE REVIEW

**Comment by Robdel12** [comment]
Looks good! The fail is related to: https://github.com/emberjs/ember.js/issues/11221 (just incase anyone stumbles upon this.)

**Comment by stefanpenner** [comment]
@Robdel12 ah ok. Is that the source of all the failures or just this one?

**Comment by Robdel12** [comment]
Yeah that's what it looks like. I'm on mobile and the Travis output is hard to read so I might have missed something :p

**Comment by stefanpenner** [comment]
> Yeah that's what it looks like. I'm on mobile and the Travis output is hard to read so I might have missed something :p

Ya I can confirm (well for stable releases, canary is another story...)

**Comment by stefanpenner** [comment]
@Robdel12 as stable failure is expected, what would you suggest the next steps should be?

**Comment by Robdel12** [comment]
@stefanpenner Man I didn't even look at canary's test fails. Glimmer did a number on it :P 

I'm not sure how to go about fixing the fails on the other versions (1.11, 1.12). Since it seems to be an ember bug It'd be nice to see that fixed so we can get this merged in and work on canary related fails. 

I'd love to help with the ember bug but I'm not sure _where_ it's busted. I dug around in the source for an hour and didn't come out with anything.

File: config/ember-try.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    {
+      name: 'ember-canary',
+      dependencies: {
+        'ember': 'components/ember#canary'
+      },
+      resolutions: {
+        'ember': 'canary'
+      }
+    }
+  ]
```
Comment (Robdel12) [inline config/ember-try.js]: Can we add Ember 1.10 to this?

**Comment by stefanpenner** [comment]
@Robdel12 added, it is also the only one to pass.

**Comment by Robdel12** [comment]
@stefanpenner I expected that. Really interested what changed between 1.10 vs 1.11 with binding attrs.

**Comment by nikz** [comment]
@Robdel12 @stefanpenner is there anything I can help out with to get this PR merged? Keen to upgrade  to ember-cli 0.2.4.

**Comment by Robdel12** [comment]
@nikz you can update to ember cli 0.2.4 but you **can't** update to Ember 1.11+ _yet_ if you are binding the `form` attribute to your select. 

There is a running bug ticket on ember that @stefanpenner and friends have been doing some great work on :) https://github.com/emberjs/ember.js/issues/11221#issuecomment-106109045

**Comment by stefanpenner** [comment]
@Robdel12 my suggestion would be to merge this. As it only accurately represents the current state. We can likely get the form fix in shortly to ember, we just have to settle on which way to solve it

**Comment by cowboyd** [comment]
How about we just conditionally skip the test on the versions that are failing.

**Comment by stefanpenner** [comment]
> How about we just conditionally skip the test on the versions that are failing.

but they are actually legitimately failing. Should we maybe move them to allowed failures?

**Comment by Robdel12** [comment]
Closing this since we've had a couple PRs that do this.

**Pull request closed without merging**